lib/core/text/flat.nit

```diff
@@ -1184,10 +1183,11 @@ redef class NativeString
 		return new FlatString.with_infos(self, len, 0)
 	end
 
-	# Returns `self` as a new String.
-	redef fun to_s_with_copy: FlatString
+	redef fun to_s_with_copy do return to_s_with_copy_and_length(cstring_length)
+
+	# Get a `String` from `length` bytes at `self` copied into Nit memory
+	fun to_s_with_copy_and_length(length: Int): String
```

Comment:
At first there was @privat : `to_s`
Then came @R4PaSs : `to_s_with_copy`
After this @xymus : `to_s_with_copy_and_length`

I want the `to_s_with_copy_and_length_and_this_null_int_that_nobody_knows_the_purpose(length: Int, foo: nullable Int)`